### PR TITLE
Feature/encryption improvements

### PR DESCRIPTION
**Pull request by f1ana**
- Stronger default hashes - DONE
- Stronger default encryption key
- Add more unit tests for hash and encryption
- NopDropdownListFor support on installation page (kind of)

If we want to allow changes in production, then we have to store hash algorith for each record in "CustomerPassword" table. Because existing records wwould be broken

### CODE REVIEW

**Comment by f1ana** [comment]
Hello,

Just checking in on the status of this pull request.  If any other changes need to be made prior to accepting, please let me know.

-Chris

**Comment by AndreiMaz** [comment]
Thanks a lot! We'll consider it a bit later. This work item (https://github.com/nopSolutions/nopCommerce/issues/295) is still open

File: src/Libraries/Nop.Core/Domain/Security/HashFormat.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nop.Core.Domain.Security {
+    public enum HashFormat {
+        Sha1 = 0,
```
Comment (ibcooley) [inline src/Libraries/Nop.Core/Domain/Security/HashFormat.cs]: I'm not a maintainer here, but why add support for SHA1?  It has been broken in practice, in some cases, and is not being used by governments any more.   I'd suggest removing this from the set of potential hash formats.

File: src/Libraries/Nop.Core/Domain/Security/EncryptionFormat.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nop.Core.Domain.Security {
+    public enum EncryptionFormat {
+        TripleDes = 0,
```
Comment (ibcooley) [inline src/Libraries/Nop.Core/Domain/Security/EncryptionFormat.cs]: I'm not a maintainer here, but why add support for TripleDes?  Isn't it much less secure than AES?

**Pull request closed without merging**